esti/lakectl_diff_test.go

```diff
@@ -0,0 +1,93 @@
+package esti
+
+import (
+	"sort"
+	"testing"
+)
+
+func TestLakectlDiffAddedFiles(t *testing.T) {
+	repoName := generateUniqueRepositoryName()
+	storage := generateUniqueStorageNamespace(repoName)
+	testBranch := "test"
+
+	createRepo(t, repoName, storage)
+	createBranch(t, repoName, storage, testBranch)
+
+	files := map[string]string{
+		"path/to/file1.txt":       "ro_1k",
+		"path/to/other/file2.txt": "ro_1k_other",
+	}
+	commitFilesForDiff(t, repoName, testBranch, files)
+	expectedDiff := &ExpectedDiff{
+		added:   []string{"path/to/file1.txt", "path/to/other/file2.txt"},
+		deleted: []string{},
+	}
+	runDiffAndExpect(t, repoName, testBranch, expectedDiff)
+}
+
+type ExpectedDiff struct {
+	added   []string
+	deleted []string
+}
+
+func (a ExpectedDiff) buildAssertionString() string {
+	type PrefixedFile struct {
+		prefix string
+		path   string
+	}
+	var added []PrefixedFile
+	var deleted []PrefixedFile
+	for _, file := range a.added {
+		added = append(added, PrefixedFile{"+ added ", file})
+	}
+	for _, file := range a.deleted {
+		deleted = append(deleted, PrefixedFile{"- deleted ", file})
+	}
+	var all = append(added, deleted...)
+	sort.Slice(all, func(i, j int) bool {
+		return all[i].path < all[j].path
+	})
+
+	var assertionString string
+	for _, file := range all {
+		assertionString += file.prefix + file.path + "\n"
+	}
+	return assertionString
+}
+
+func runDiffAndExpect(t *testing.T, repoName string, testBranch string, diffArgs *ExpectedDiff) {
+	diffVars := map[string]string{
+		"REPO":         repoName,
+		"LEFT_BRANCH":  mainBranch,
+		"RIGHT_BRANCH": testBranch,
+		"DIFF_LIST":    diffArgs.buildAssertionString(),
+	}
+	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" diff lakefs://"+repoName+"/"+mainBranch+" lakefs://"+repoName+"/test", false, "lakectl_diff", diffVars)
+}
+
+func commitFilesForDiff(t *testing.T, repoName string, branch string, files map[string]string) {
+	for filePath, contentPath := range files {
+		RunCmdAndVerifyContainsText(t, Lakectl()+" fs upload -s files/"+contentPath+" lakefs://"+repoName+"/"+branch+"/"+filePath, false, filePath, nil)
+	}
+
+	commitMessage := "committing test files to " + branch
+	RunCmdAndVerifyContainsText(t, Lakectl()+" commit lakefs://"+repoName+"/"+branch+" -m \""+commitMessage+"\"", false, commitMessage, nil)
+}
```

Comment:
dear reviewer, please lmk wdyt about the following:

when i try to assert with the expected golden file it fails with this message:
![image](https://github.com/treeverse/lakeFS/assets/51454184/1603e91d-b103-4a3b-9212-1fc31b406a6a)
and since i don't test the upload or commit functionality here, i figured it's ok to have a weaker assertion on the result, hence the usage of `RunCmdAndVerifyContainsText` without vars.